packages/next/src/client/dev/dev-build-indicator/internal/handle-dev-build-indicator-hmr-events.ts

```diff
@@ -0,0 +1,25 @@
+import { HMR_ACTIONS_SENT_TO_BROWSER } from '../../../../server/dev/hot-reloader-types'
+import { devBuildIndicator } from './dev-build-indicator'
+
+/**
+ * Handles HMR events to control the dev build indicator visibility.
+ * Shows indicator when building and hides it when build completes or syncs.
+ */
+export const handleDevBuildIndicatorHmrEvents = (obj: any) => {
```

Comment:
can this be typed with [`HMR_ACTION_TYPES`](https://github.com/vercel/next.js/blob/58cd123fc8910b670cfab9938ad0fc94c7fb7cb6/packages/next/src/server/dev/hot-reloader-types.ts#L118)?